### PR TITLE
Change tests according to ecma402/pull/278

### PR DESCRIPTION
**Pull request by FrankYFTang**
per https://github.com/tc39/ecma402/pull/278 remove the harden retirement for array returned by supportLocalesOf
@littledan @Ms2ger @gsathya @zbraniecki @anba

### CODE REVIEW

File: test/intl402/ListFormat/constructor/supportedLocalesOf/result-type.js [hunk 1]
```diff
@@ -21,14 +21,14 @@ assert.sameValue(Object.isExtensible(result), true,
...
+    "writable": true,
     "enumerable": true,
-    "configurable": false,
+    "configurable": true,
   });
 }
 
 verifyProperty(result, "length", {
-  "writable": false,
+  "writable": true,
```
Comment (jakobkummerow) [inline test/intl402/ListFormat/constructor/supportedLocalesOf/result-type.js]: It looks like `verifyProperty` is too limited to successfully check writability of Array.length, so I'd just drop this line.

Comment (FrankYFTang) [reply test/intl402/ListFormat/constructor/supportedLocalesOf/result-type.js]: Thanks. I update the PR. PTAL. Thanks

**Comment by FrankYFTang** [comment]
@littledan - is this PR good enough to complete your request in "Will you write test262 tests if I merge this?" in https://github.com/tc39/ecma402/pull/278 ?

File: test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js [hunk 2]
```diff
@@ -21,14 +21,13 @@ assert.sameValue(Object.isExtensible(result), true,
...
-    "writable": false,
+    "writable": true,
     "enumerable": true,
-    "configurable": false,
+    "configurable": true,
   });
 }
 
 verifyProperty(result, "length", {
-  "writable": false,
```
[the remark this answers is not in the document]
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: The verifyProperty function is pretty buggy now. IF you can fix it, I can added it back. According to jakobkummerow 3 hours ago
> It looks like verifyProperty is too limited to successfully check writability of Array.length, so I'd just drop this line.

[the remark this answers is not in the document]
Comment (leobalter) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: Would you explain what's exactly the bug here or how is it limited?

File: test/intl402/supportedLocalesOf-returned-array-elements-are-not-frozen.js [hunk 3]
```diff
@@ -5,14 +5,20 @@
...
+function testNormalProperty(obj, property) {
     var desc = Object.getOwnPropertyDescriptor(obj, property);
-    assert.sameValue(desc.writable, false, "Property " + property + " of object returned by SupportedLocales is writable.");
+    assert.sameValue(desc.writable, true, "Property " + property + " of object returned by SupportedLocales is not writable.");
+    assert.sameValue(desc.configurable, true, "Property " + property + " of object returned by SupportedLocales is not configurable.");
+}
+
+function testLengthProperty(obj, property) {
+    var desc = Object.getOwnPropertyDescriptor(obj, property);
+    assert.sameValue(desc.writable, true, "Property " + property + " of object returned by SupportedLocales is not writable.");
```
Comment (leobalter) [inline test/intl402/supportedLocalesOf-returned-array-elements-are-not-frozen.js]: the negative messages for the assertions are really confusing. I see that `desc.writable` should be `true` and then a message saying it's not writable. Assertions messages shouldn't be error messages, but a description of each assertion.

This doesn't need to be changed in this PR, as this file already exists, but it would be nicer in the future PRs to follow a convention.

File: test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: https://tc39.github.io/ecma262/#sec-property-attributes said
"
[[Writable]] | Boolean | If false, attempts by ECMAScript code to change the property's [[Value]] attribute using [[Set]] will not succeed.
"
Notice it does not stated "throw TypeError" but "not succeed". I somehow got a "RangeError" (which is "not succeed") but not TypeError. 

But in isWritable the function inside harness/propertyHelper.js which check the writable or not, it is coded specifically checking for TypeError and won't consider other kind of "not succeded".  Even if I change to include RangeError, I also hit some issue (I cannot figure out what exactly) with "length".

[the remark this answers is not in the document]
Comment (FrankYFTang) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: @ jakobkummerow

[the remark this answers is not in the document]
Comment (jakobkummerow) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: When a `"writable": true` property is in the options object, then `verifyProperty` [calls](https://github.com/tc39/test262/blob/master/harness/propertyHelper.js#L64) `isWritable(array, "length")`, which in absense of the optional 4th argument will [try](https://github.com/tc39/test262/blob/master/harness/propertyHelper.js#L129) to set `array["length"] = "unlikelyValue"`, which throws a `RangeError`, which causes `isWritable` to throw an exception because it will only accept TypeErrors there. Even if that is fixed, it will then [check](https://github.com/tc39/test262/blob/master/harness/propertyHelper.js#L136) whether `array.length === "unlikelyValue"` after the assignment, which will of course never be true, even though the property is, in fact, writable.

In short, `verifyProperty` in its current form cannot be used to assert writability of `Array.length`, and it is clearly beyond the scope of this patch to fix that.

A workaround would be to do a separate `verifyWritable(result, "length")` call after the writable-ignoring `verifyProperty` call.

[the remark this answers is not in the document]
Comment (littledan) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: The difficulty of using the property verification methods has been raised by many different test262 contributors. Maybe the testing should be based on getOwnPropertyDescriptor by default, and the more intense testing can be used in cases where we're especially suspicious of a potential issue (e.g., in RegExp legacy properties).

[the remark this answers is not in the document]
Comment (leobalter) [reply test/intl402/Segmenter/constructor/supportedLocalesOf/result-type.js]: By the reasons pointed by @jakobkummerow, the writable check doesn't work for `Array#length` (thanks for clarifying that, btw). Although, a verifyWritable call would have the same result and issue.

In this case it would be nice to just have a simple check for the writable attr in the descriptor of this case. We might discuss other issues elsewhere not blocking this PR.
